compiler/circle-opselector/driver/Driver.cpp

```diff
@@ -0,0 +1,306 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "OpSelector.h"
+// TODO Add new pass headers
+
+#include <foder/FileLoader.h>
+
+#include <luci/Importer.h>
+#include <luci/CircleExporter.h>
+#include <luci/CircleFileExpContract.h>
+#include <luci/Import/CircleReader.h>
+
+#include <arser/arser.h>
+#include <vconone/vconone.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+void print_version(void)
+{
+  std::cout << "circle-opselector version " << vconone::get_string() << std::endl;
+  std::cout << vconone::get_copyright() << std::endl;
+}
+
+bool check_input(std::string str)
+{
+  bool check_hyphen = false;
+
+  for (char c : str)
+  {
+    if ('0' <= c && c <= '9')
+      continue;
+    else if (check_hyphen) // when user enter '-' more than 2.
+    {
+      std::cout << "Too many '-' in str." << std::endl;
+      exit(0);
+    }
+    else if (c == '-')
+      check_hyphen = true;
+    else // when user enter not allowed character, print alert msg.
+    {
+      std::cout << "To select operator by id, please use these args: [0-9], '-', ','" << std::endl;
+      exit(0);
+    }
+  }
+  return true;
+}
+
+void split_id_input(const std::string &str, std::vector<int> &by_id)
+{
+  std::istringstream ss;
+  ss.str(str);
+  std::string str_buf;
+
+  while (getline(ss, str_buf, ','))
+  {
+    if (str_buf.length() && check_input(str_buf)) // input validation
+    {
+      if (str_buf.find('-') == std::string::npos) // if token has no '-'
+        by_id.push_back(stoi(str_buf));
+      else // tokenize again by '-'
+      {
+        std::istringstream ss2(str_buf);
+        std::string token;
+        int from_to[2], top = 0;
+
+        while (getline(ss2, token, '-'))
+          from_to[top++] = stoi(token);
+
+        for (int number = from_to[0]; number <= from_to[1]; number++)
+          by_id.push_back(number);
+      }
+    }
+  }
+}
+
+void split_name_input(const std::string &str, std::vector<std::string> &by_name)
+{
+  std::istringstream ss;
+  ss.str(str);
+  std::string str_buf;
+
+  while (getline(ss, str_buf, ','))
+    by_name.push_back(str_buf);
+}
+
+void getOplist(std::string op, std::vector<int> &oplist)
+{
+  char op_buf[1000];
+  strcpy(op_buf, op.c_str());
+  char *op_next = op_buf;
+
+  // tokenize by ,
+  char *tok_comma = strtok_r(op_buf, ",", &op_next);
+  while (tok_comma != nullptr)
+  {
+    char buf[100];
+    strcpy(buf, tok_comma);
+    char *buf_next = buf;
+
+    if (isdigit(tok_comma[0]))
```

Comment:
@Paran-Yu Does `--select` and `--deselect` can accept both `id` and `name` by checking whether it is `digit` or not?

If it is valid assumption, we may remove `--by-id` and `--by-name` option also.
It it is not guaranteed, we need some way to give explicitly whether it is selecting by id or selecting by name.

As I understand no one guarantees the `id` will be digit number, and `name` will be non-digit.

Reply:
Options `--select` and `--deselect` can accept only `id` now, but we think it is not efficient.
So we would modify options like this. #7667